app/products/calls/screens/call_screen/call_screen.tsx

```diff
@@ -557,6 +557,13 @@ const CallScreen = ({
             // ignore the error because the call screen has likely already been popped async
             Navigation.pop(componentId).catch(() => null);
         }
+
+        // // If we end a call and return from the Thread screen, the "visible screen" is still the thread.
+        if (NavigationStore.getVisibleScreen() === Screens.THREAD) {
```

Comment:
What if we are looking at a permalink? Or we are editing a post? That would be one extra screen on the stack.

And that would go along the proposals from Elias. If I am editing a post, I don't want to lose what I have edited because the call finished.

Reply:
> What if we are looking at a permalink? Or we are editing a post? That would be one extra screen on the stack.
> 
> And that would go along the proposals from Elias. If I am editing a post, I don't want to lose what I have edited because the call finished.

Those screens won't be popped by this PR's code, so you won't lose your work or the screen you're on. The only screen that gets popped on its own is the thread screen when it is the direct child of the Call screen (meaning: the user pressed the `Call Thread` button in the Call screen).

But, since there seems to be confusion about what this code does, I think that indicates it could be clarified. I took a look at it and I found a better way to pop the Call screen. I'm removing the code that was scattered around and moving it to the `myselfLeftCall` function (DRY and all that). I simplified the popping to just calling the `Navigation.pop(Screens.CALL)` since it does the same thing, and it looks like those are noops if the screen isn't on the stack. Let me know what you think.

Unfortunately the thread screen still gets popped like before, if it's the direct child of the Call screen. I'm not sure if there's a way to prevent that. But we can look at that more deeply in https://mattermost.atlassian.net/browse/MM-52971

Reply:
Just to be sure, this is the steps I am worried about:
- Start a call
- Click the call widget, so you get into the call screen
- Hit more and Call Thread to go to the thread
- Publish one message
- Long press on that message
- Hit edit

This way, the stack may easily be...
Home -> Channel -> Thread -> Call -> Call Thread -> Edit

If the call finishes then, what happens? (I guess the call can finish without me pushing the button, for example by issues on the network)

Reply:
Also, I have seen in other places that not properly dismissing the bottom sheet may cause issues, so it would be interesting to verify that if finishing the call while the bottom sheet is open creates any issue. And also while attaching an image (it is a weird scenario because you are moving to the gallery to select the image).

Reply:
I've tested it, it doesn't pop the edit screen. I also tried the permalink case (which for my local server brings up the browser), and that's fine too. Are there any other cases you're interested in?